src/Raven.Server/Utils/Monitoring/ServerMetrics.cs

```diff
@@ -0,0 +1,234 @@
+        public AllDatabasesMetrics Databases { get; set; }
+        
+        public DynamicJsonValue ToJson()
+        {
+            return new DynamicJsonValue
+            {
+                [nameof(ServerVersion)] = ServerVersion,
+                [nameof(ServerFullVersion)] = ServerFullVersion,
+                [nameof(UpTimeInSec)] = UpTimeInSec,
+                [nameof(ServerProcessId)] = ServerProcessId,
+                [nameof(CurrentNumberOfRunningBackups)] = CurrentNumberOfRunningBackups,
+                [nameof(Config)] = Config.ToJson(),
+                [nameof(Cpu)] = Cpu.ToJson(),
+                [nameof(Memory)] = Memory.ToJson(),
+                [nameof(Disk)] = Disk.ToJson(),
+                [nameof(License)] = License.ToJson(),
+                [nameof(Network)] = Network.ToJson(),
+                [nameof(Certificate)] = Certificate.ToJson(),
+                [nameof(Cluster)] = Cluster.ToJson(),
+                [nameof(Databases)] = Databases.ToJson()
+            };
+        }
+    }
+
+    public class ConfigurationMetrics
+    {
+        public string[] Urls { get; set; }
+        public string PublicUrl { get; set; }
+        public string[] TcpUrls { get; set; }
+        public string[] PublicTcpUrls { get; set; }
+        public int MaxNumberOfConcurrentBackups { get; set; }
+
+        public DynamicJsonValue ToJson()
+        {
+            return new DynamicJsonValue
+            {
+                [nameof(Urls)] = Urls,
+                [nameof(PublicUrl)] = PublicUrl,
+                [nameof(TcpUrls)] = TcpUrls,
+                [nameof(PublicTcpUrls)] = PublicTcpUrls,
+                [nameof(MaxNumberOfConcurrentBackups)] = MaxNumberOfConcurrentBackups
+            };
+        }
+    }
+
+    public class CpuMetrics
+    {
+        public double ProcessUsage { get; set; }
+        public double MachineUsage { get; set; }
+        public double? MachineIoWait { get; set; }
+        public int ProcessorCount { get; set; }
+        public int AssignedProcessorCount { get; set; }
+        public int ThreadPoolAvailableWorkerThreads { get; set; }
+        public int ThreadPoolAvailableCompletionPortThreads { get; set; }
+
+        public DynamicJsonValue ToJson()
+        {
+            return new DynamicJsonValue
+            {
+                [nameof(ProcessUsage)] = ProcessUsage,
+                [nameof(MachineUsage)] = MachineUsage,
+                [nameof(MachineIoWait)] = MachineIoWait,
+                [nameof(ProcessorCount)] = ProcessorCount,
+                [nameof(AssignedProcessorCount)] = AssignedProcessorCount,
+                [nameof(ThreadPoolAvailableWorkerThreads)] = ThreadPoolAvailableWorkerThreads,
+                [nameof(ThreadPoolAvailableCompletionPortThreads)] = ThreadPoolAvailableCompletionPortThreads
+            };
+        }
+    }
+
+    public class MemoryMetrics
+    {
+        public long TotalMemoryInMb { get; set; }
+        public bool LowState { get; set; }
```

Comment:
LowState?

I think we also have 3 states now, Low, ExtremlyLow, Normal?

Reply:
ok will check that and replace with enum.

Reply:
changed to: LowMemorySeverity